### PR TITLE
Updating Lucene from Version 4.10.3 to 7.4.0

### PR DESCRIPTION
**Pull request by Awti**
The jar still contains the old Lucene files (4.10.3), if this rolls out smoothly they can be removed

### CODE REVIEW

File: src/main/java/suneido/runtime/builtin/Lucene.java [hunk 1]
```diff
@@ -59,15 +61,19 @@ protected Object newInstance(Object... args) {
...
+		try	(FSDirectory dir = FSDirectory.open(getPath(a));
 				DirectoryReader reader = DirectoryReader.open(dir)) {
 			new IndexSearcher(reader);
 			return true;
 		} catch (IOException e) {
 			return false;
 		}
 	}
+	
+	@Params("dir")
```
Comment (apmckinlay) [inline src/main/java/suneido/runtime/builtin/Lucene.java:73]: @Params is only needed for methods that are callable from Suneido code.

File: src/main/java/suneido/runtime/builtin/Lucene.java [hunk 2]
```diff
@@ -103,8 +109,7 @@ public static Object Insert(Object self, Object key, Object text) {
 			doc.add(new StringField("key", Ops.toStr(key), Field.Store.YES));
 
 			FieldType type = new FieldType();
-	        type.setIndexed(true);
```
Comment (apmckinlay) [inline src/main/java/suneido/runtime/builtin/Lucene.java:106]: setIndexed isn't needed any more?

File: src/main/java/suneido/runtime/builtin/Lucene.java [hunk 3]
```diff
@@ -153,9 +158,9 @@ void close() {
 
 	static IndexWriter writer(String path, boolean create) {
 		try {
-			Directory dir = FSDirectory.open(new File(path));
+			Directory dir = FSDirectory.open(getPath(path));
```
Comment (apmckinlay) [inline src/main/java/suneido/runtime/builtin/Lucene.java:161]: This getPath could just be Paths.get since path is already a string

File: src/main/java/suneido/runtime/builtin/Lucene.java [hunk 4]
```diff
@@ -177,13 +181,14 @@ public static Object Search(Object self, Object a, Object b, Object c, Object d)
 			TopDocs results = searcher.search(query, limit);
 			SimpleHTMLFormatter htmlFormatter = new SimpleHTMLFormatter();
 			Highlighter highlighter = new Highlighter(htmlFormatter, new QueryScorer(query));
+			String key = "";
```
Comment (apmckinlay) [inline src/main/java/suneido/runtime/builtin/Lucene.java:184]: Why did you move this out of the loop?

File: src/main/java/suneido/runtime/builtin/Lucene.java (hunk 3 above)
Comment (Awti) [reply src/main/java/suneido/runtime/builtin/Lucene.java:161]: It could be. I simply did it to standardize the calling of Paths. Even though it is a string it is handled the same

File: src/main/java/suneido/runtime/builtin/Lucene.java (hunk 4 above)
Comment (Awti) [reply src/main/java/suneido/runtime/builtin/Lucene.java:184]: Preventative measure. If there is nothing to loop through, key does not get initialized yet it is used following the loop

File: src/main/java/suneido/runtime/builtin/Lucene.java (hunk 1 above)
Comment (Awti) [reply src/main/java/suneido/runtime/builtin/Lucene.java:73]: Ah, I will clean that up then

File: src/main/java/suneido/runtime/builtin/Lucene.java (hunk 2 above)
Comment (Awti) [reply src/main/java/suneido/runtime/builtin/Lucene.java:106]: It isn't even a method of field type anymore. Unless specified to not have an index, it seems to be indexed

File: src/main/java/suneido/runtime/builtin/Lucene.java (hunk 4 above)
Comment (apmckinlay) [reply src/main/java/suneido/runtime/builtin/Lucene.java:184]: I don't see it used outside the loop. Java wouldn't allow that.

Comment (Awti) [reply src/main/java/suneido/runtime/builtin/Lucene.java:184]: I'll double check and correct when I am back in tomorrow. Could have sworn it was passed into Ops.call1 but I must be mistaken

**Comment by apmckinlay** [comment]
yes, but that's inside the loop

**Comment by Awti** [comment]
Ahhh to much time in suneido, my levels of indentation are all off. I'll correct that with the other changes and send again tomorrow morning
